apps/src/templates/rubrics/teacherRubricRedux.ts

```diff
@@ -111,7 +122,7 @@ export const loadAllTeacherEvaluationData = createAsyncThunk(
       allTeacherEvaluationData: AllTeacherEvaluationData
     ) => {
       const hasFeedbackMap: HasTeacherFeedbackMap = {};
-      allTeacherEvaluationData.forEach(userEvalData => {
+      allTeacherEvaluationData?.forEach(userEvalData => {
```

Comment:
Initial state sets `allTeacherEvaluationData` to an empty array. The type declaration doesn't allow nulls. Not sure why the `?` is needed here?

Reply:
Good catch. I added this to address some inconsistent unit test failures in this test case: https://github.com/code-dot-org/code-dot-org/blob/c2b4dd8befccbc16fa6c495e85c4e0bf21101cc7/apps/test/unit/templates/rubrics/RubricFloatingActionButtonTest.jsx#L183

however your reasoning prompted me to consider that this might come from not having mocked out fetch calls to return valid values. fixed in 4f612968ea928cf3d44c100f59dbc183acc86f3c

Reply:
With this new understanding, I'm now able to repro fairly consistently a test failure when there is any test in this file that has the `sectionId` prop set (necessary for network calls to be issued), and does not call `stubFetch`.